simplified-main/src/main/java/org/nypl/simplified/main/MainFragmentViewModel.kt

```diff
@@ -150,19 +150,24 @@ class MainFragmentViewModel(
           )
 
         getHoldsFuture = this.profilesController.profileFeed(request)
-        val numberOfHolds = getHoldsFuture
-          ?.get()
-          ?.entriesInOrder
-          .orEmpty()
-          .filter { feedEntry ->
-            feedEntry is FeedEntry.FeedEntryOPDS &&
-              feedEntry.feedEntry.availability is OPDSAvailabilityHeldReady
-          }
-          .size
-
-        bookRegistry.updateHolds(
-          numberOfHolds = numberOfHolds
-        )
+
+        try {
+          val numberOfHolds = getHoldsFuture
+            ?.get()
+            ?.entriesInOrder
```

Comment:
On what thread does this call happen?

Reply:
The reason I ask is that it's rarely a good idea to call `get()` on a future directly as it's a blocking call. If this happens on the UI thread... That's bad.

Reply:
This is running on a background thread, but the `initializeHoldsCallTimer()` was being called while that call was running and since it interrupts it, the app was crashing because of a `InterruptedException`

Reply:
That's true, but the `getHoldsFuture` variable is declared as nullable and that's why the `?` are being used.

Reply:
Ah, I see. I didn't realize it was declared above as a field in the class.

I don't like how this viewmodel is structured, but this PR doesn't make it any worse!

Reply:
Refactoring time is always a good time :D